lib/honeybadger/insights/base.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Honeybadger.Insights.Base do
+  @moduledoc """
+  Base module providing common telemetry attachment functionality.
+  """
+
+  defmacro __using__(_opts) do
+    quote do
+      require Logger
+
+      # Define empty defaults for the attributes
+      Module.register_attribute(__MODULE__, :required_dependencies, [])
+      Module.register_attribute(__MODULE__, :telemetry_events, [])
+
+      @before_compile Honeybadger.Insights.Base
+    end
+  end
+
+  defmacro __before_compile__(env) do
+    namespace =
+      env.module
+      |> Atom.to_string()
+      |> String.split(".")
+      |> List.last()
+      |> Macro.underscore()
+      |> String.to_atom()
+
+    quote do
```

Comment:
This generates a lot of content for each module that's paramaterizable. Usually, the preferred approach is to generate wrappers that call functions on the parent module. For example,

```elixir
def dependencies_available? do
  Base.dependencies_available?(@required_dependencies)
end
```

Reply:
Is this more what you were thinking about?

https://github.com/honeybadger-io/honeybadger-elixir/pull/591/commits/0ad97009fda823273a0d15acb01d9d44951470eb